concrete/src/Entity/File/Version.php

```diff
@@ -945,15 +948,20 @@ public function getDetailThumbnailImage()
      */
     public function getThumbnailURL($type)
     {
+        $app = Application::getFacadeApplication();
+        $canonicalUrl = $app->make('url/canonical');
+
         if (!($type instanceof ThumbnailTypeVersion)) {
             $type = ThumbnailTypeVersion::getByHandle($type);
         }
 
         /** @var Resolver $path_resolver */
-        $path_resolver = Core::make('Concrete\Core\File\Image\Thumbnail\Path\Resolver');
+        $path_resolver = $app->make('Concrete\Core\File\Image\Thumbnail\Path\Resolver');
 
         if ($path = $path_resolver->getPath($this, $type)) {
-            return $path;
+            $urlPath = $canonicalUrl->getPath();
```

Comment:
This is breaking by S3 addon, reverting this to `return $path;` fixes it, and seems to work fine with the local storage, @KorvinSzanto is making this change ok?